src/test/java/org/apache/commons/io/input/BrokenReaderTest.java

```diff
@@ -20,70 +20,125 @@
 import static org.junit.jupiter.api.Assertions.assertNotNull;
 import static org.junit.jupiter.api.Assertions.assertThrows;
 
+import java.io.FileNotFoundException;
 import java.io.IOException;
 import java.io.Reader;
+import java.nio.file.FileSystemNotFoundException;
+import java.util.stream.Stream;
 
-import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
 
 /**
  * Tests {@link BrokenReader}.
  */
 public class BrokenReaderTest {
 
-    private IOException exception;
+    static final class CustomException extends Exception {
 
-    private Reader brokenReader;
+        private static final long serialVersionUID = 1L;
 
-    @BeforeEach
-    public void setUp() {
-        exception = new IOException("test exception");
-        brokenReader = new BrokenReader(exception);
     }
 
-    @Test
-    public void testClose() {
-        assertEquals(exception, assertThrows(IOException.class, () -> brokenReader.close()));
+    static Stream<Class<? extends Throwable>> parameters() {
```

Comment:
Let's not duplicate this factory method all over the place. Extract this method in a new class `TestFixtures' or some such name.